python/fastsim/utils/utilities.py

```diff
@@ -117,6 +117,28 @@ def set_log_level(level: str | int) -> int:
         Previous log level
     """
     # Map string name to logging level
+
+    allowed_args = [
+        "CRITICAL",
+        50,
+        "ERROR",
+        40,
+        "WARNING",
+        30,
+        "INFO",
+        20,
+        "DEBUG",
+        10,
+        "NOTSET",
+        0,
+        # no logging of anything ever!
+        logging.CRITICAL + 1,
+    ]
+
+    err_str = f"Invalid arg: '{level}'.  See doc string:\n{set_log_level.__doc__}"
+
+    assert level in allowed_args, err_str
+
     if isinstance(level, str):
         level = logging._nameToLevel[level]
```

Comment:
@calbaker optionally:
```suggestion
        level = logging._nameToLevel[level.upper()]
```
This would allow lowercase log level strings `"warning"`, `"info"`, etc., but we would need to add those to the allowable inputs.

Reply:
@kylecarow , yea that sounds good.  Could you make that change?